auth/token-managers/iam-request-based-token-manager.ts

```diff
@@ -180,4 +181,24 @@ export class IamRequestBasedTokenManager extends JwtTokenManager {
 
     return this.requestWrapperInstance.sendRequest(parameters);
   }
+
+  /**
+   * Returns true iff the currently-cached IAM access token is expired.
+   * We'll consider an access token as expired when we reach its IAM server-reported
+   * expiration time minus our expiration window (10 secs).
+   * We do this to avoid using an access token that might expire in the middle of a long-running
+   * transaction within an IBM Cloud service.
+   *
+   * @returns true if the token has expired, false otherwise
+   */
+  protected isTokenExpired(): boolean {
```

Comment:
This is the key part of the fix. We'll override the inherited `isTokenExpired()` function so that it returns true as soon as we get within 10 seconds of
the IAM server-reported expiration time.